pkg/kcp/provider/gcp/vpcpeering/deleteVpcPeering.go

```diff
@@ -25,6 +25,9 @@ func deleteVpcPeering(ctx context.Context, st composed.State) (error, context.Co
 	)
 
 	if err != nil {
+		if composed.IsMarkedForDeletion(state.ObjAsVpcPeering()) {
+			return nil, nil
+		}
```

Comment:
This is unnecessary, this block is only entered if the object is marked for deletion